os/kernel/task/task_setup.c

```diff
@@ -449,6 +452,13 @@ static int thread_schedsetup(FAR struct tcb_s *tcb, int priority, start_t start,
 
 		up_initial_state(tcb);
 
+#ifdef CONFIG_APP_BINARY_SEPARATION
+		/* Copy the parent task ram details to this task */
+		rtcb = this_task();
```

Comment:
If rtcb task is not created by exec, ram_start and ram_size is not assigned.
Is it okay?

Reply:
This change is used to propagate the ram_start and ram_size from parent task to child task. This will happen whenever a new task is created (using task_create or pthread_create or exec). However, in case of exec, we are overwriting the values of ram_start and ram_size according to the actual values for the new app being launched. This overwrite is performed in the exec API flow.